### PR TITLE
[Coral-Service] Add rewriteType field to translation endpoint

### PR DESCRIPTION
**Pull request by KevinGe00**
### What changes are proposed in this pull request, and why are they necessary?

Adding a rewrite field in the `api/translations/translate` endpoint to support new coral-service rewrite feature. This PR also modifies coral service to accept Spark as a source language.

### How was this patch tested?
no rewrite param specified, returns a translation with no rewrite
```
curl --header "Content-Type: application/json" \
  --request POST \
  --data '{
    "fromLanguage":"hive", 
    "toLanguage":"trino", 
    "query":"SELECT * FROM db1.airport"
  }' \
  http://localhost:8080/api/translations/translate`
```
rewrite param is "none", returns a translation with no rewrite
```
curl --header "Content-Type: application/json" \
    --request POST \                                                                                                              
    --data '{                                  
    "sourceLanguage":"hive",
    "targetLanguage":"trino",
    "query":"SELECT * FROM db1.airport",
    "rewriteType":"none"
    }' \
```

valid rewrite param, returns translation with requested rewrite
```
curl --header "Content-Type: application/json" \
    --request POST \                                                                                                              
    --data '{                                  
    "sourceLanguage":"hive",
    "targetLanguage":"trino",
    "query":"SELECT * FROM db1.airport",
    "rewriteType":"incremental"
    }' \
```

invalid rewrite param, returns a translation with no rewrite
```
curl --header "Content-Type: application/json" \
    --request POST \                                                                                                              
    --data '{                                  
    "sourceLanguage":"hive",
    "targetLanguage":"trino",
    "query":"SELECT * FROM db1.airport",
    "rewriteType":"notavalidrewritetype"
    }' \
```

### CODE REVIEW

File: coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java [hunk 1]
```diff
@@ -59,32 +61,35 @@ public ResponseEntity translate(@RequestBody TranslateRequestBody translateReque
...
 
+    // TODO: Allow translations between the same language
     if (sourceLanguage.equalsIgnoreCase(targetLanguage)) {
       return ResponseEntity.status(HttpStatus.BAD_REQUEST)
           .body("Please choose different languages to translate between.\n");
     }
 
+    if (!isValidSourceLanguage(sourceLanguage)) {
+      return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+          .body("Currently, only Hive, and Trino are supported as source/input languages.\n");
```
Comment (wmoustafa) [inline coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java:74]: no need for `/input`.

File: coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java [hunk 2]
```diff
@@ -117,6 +122,7 @@ public ResponseEntity getIncrementalInfo(@RequestBody IncrementalRequestBody inc
     incrementalResponseBody.setIncrementalQuery(null);
     try {
       if (language.equalsIgnoreCase("spark")) {
+        // TODO: replace getSparkIncrementalQueryFromUserSql with getIncrementalQuery
         String incrementalQuery = getSparkIncrementalQueryFromUserSql(query);
```
Comment (wmoustafa) [inline coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java:126]: How does this work when the source/target specified by the user is not Spark?

File: coral-service/src/main/java/com/linkedin/coral/coralservice/controller/VisualizationController.java [hunk 3]
```diff
@@ -43,9 +44,9 @@ public ResponseEntity getIRVisualizations(@RequestBody VisualizationRequestBody
     final String query = visualizationRequestBody.getQuery();
     final RewriteType rewriteType = visualizationRequestBody.getRewriteType();
 
-    if (!visualizationUtils.isValidSourceLanguage(sourceLanguage)) {
+    if (!isValidSourceLanguage(sourceLanguage)) {
       return ResponseEntity.status(HttpStatus.BAD_REQUEST)
-          .body("Currently, only Hive, Spark, and Trino are supported as engines to generate graphs using.\n");
+          .body("Currently, only Hive, and Trino are supported as engines to generate graphs using.\n");
```
Comment (wmoustafa) [inline coral-service/src/main/java/com/linkedin/coral/coralservice/controller/VisualizationController.java:49]: Let us allow Spark and route it to Hive. Also you can say
`Currently, only Hive, Spark, and Trino SQL are supported as source languages for Coral IR visualization`

File: coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java (hunk 2 above)
Comment (KevinGe00) [reply coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java:126]: `getIncrementalQuery(String query, String sourceLanguage, String targetLanguage)`  takes in a sourceLanguage param.

File: coral-service/src/main/java/com/linkedin/coral/coralservice/controller/VisualizationController.java (hunk 3 above)
Comment (KevinGe00) [reply coral-service/src/main/java/com/linkedin/coral/coralservice/controller/VisualizationController.java:49]: Do we also do this for the translation endpoint?

File: coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java (hunk 2 above)
Comment (wmoustafa) [reply coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java:126]: Still not quite clear. Why are not we using `getIncrementalQuery` here and leaving it as a TODO?

Comment (wmoustafa) [reply coral-service/src/main/java/com/linkedin/coral/coralservice/controller/TranslationController.java:126]: Can we hardcode "spark" and get rid of `getSparkIncrementalQueryFromUserSql`? The TODO will look different as well.

**Review by wmoustafa — APPROVED** [review]

**Pull request merged**